lib/httparty/request.rb

```diff
@@ -276,11 +276,10 @@ def get_charset
     end
 
     def encode_with_ruby_encoding(body, charset)
-      if !body.nil? && Encoding.name_list.include?(charset)
-        body.force_encoding(charset)
-      else
-        body
-      end
+      encoding = Encoding.find(charset)
+      body.force_encoding(charset)
+    rescue ArgumentError
```

Comment:
Ohhhhh, this is a mistake; I had assumed the test changes in c46663d73ff96d86b92d36208a021bbe58672e7c covered the `body.nil?` case that was added to this method in the same commit and then didn’t worry about this because the tests passed.

Turns out they don’t; I’ll add a test for that and put the explicit `body.nil?` check back in.

(FWIW, I forgot the `ArgumentError` type check wouldn’t cover the `body.nil?` case and was thinking leaving out the explicit `nil` check made the code a little simpler. D'oh!)